src/main/python/community-detector/community_detector.py

```diff
@@ -0,0 +1,154 @@
+from collections import defaultdict
+from itertools import chain
+import logging
+
+from igraph import Graph
+import numpy
+
+
+def detect_communities(cc,
+                       buckets_matrix,
+                       edges="linear",
+                       algorithm="walktrap",
+                       algorithm_params={}):
+    """Runs Community Detection analysis on the given connected components.
+
+    Based largely on the Apollo detect_communities() code from
+    https://github.com/src-d/apollo/blob/master/apollo/graph.py
```

Comment:
Grr, GH is not showing reviews from commits in the PR https://github.com/src-d/gemini/commit/6a1993492f49f47e83c211d2d7bf6d90260b3444#r28488078 :/